### PR TITLE
Integrated dark/light mode icon with fully working functionality

### PR DESCRIPTION
**Pull request by kumartushar0605**
I have added a dark/light mode toggle icon for easy switching between themes. It’s fully functional, offering smooth transitions and remembering your preference across pages. The icon changes (e.g., sun/moon) based on the active mode, enhancing user experience with real-time feedback.
Screenshot (170)

### CODE REVIEW

File: client/package.json
Comment (PorePranav) [inline client/package.json:14]: Please remove all unused libraries from the project.

File: client/src/main.jsx
Comment (PorePranav) [inline client/src/main.jsx:11]: Please export the context to it's individual file and then import it into the main file to maintain readability

File: client/src/pages/Cart.jsx
Comment (PorePranav) [inline client/src/pages/Cart.jsx:23]: Avoid using style on all page components to ensure proper maintainability. Make use of tailwind classes instead.

**Review by PorePranav — CHANGES_REQUESTED** [review]
Thank you for your contribution to the CampusUnify project. Here are some points to address before we can merge this PR:

## Dark Mode Implementation

To properly implement dark mode, please use Tailwind classes and colors instead of applying classes and colors directly to page components. This approach ensures consistency and makes it easier to maintain the codebase. I know that this would be a overwhelming task, If you need help, I can assign more people.

## Unused Libraries
Please remove all unused libraries from the project before submitting the PR.

## Code Quality

Before submitting PRs, please ensure you run the following commands:

```
npm run lint
npm run format
```

This helps maintain code quality and consistency across the project.

## Project Coding Style

Please take a moment to review our [CONTRIBUTING guidelines](https://github.com/PorePranav/CampusUnify/blob/main/CONTRIBUTING.md). It contains important information about our project's coding style and guidelines. Please separate out the context for DarkMode into a separate file to help with code readability. Strictly use TailwindCSS for styling rather than inline styles.

## Color Issues

Some colors in the current PR aren't working correctly. Please review and fix these issues.
For example, check the below images:

Screenshot 2024-10-04 224711
image


## Line-by-Line Comments

For more specific feedback, please refer to the line-by-line comments in the PR review. These comments provide detailed suggestions and point out areas that need attention.

## Action Items

1. Refactor dark mode implementation using Tailwind classes and colors.
2. Run linting and formatting scripts.
3. Review and adhere to the coding style outlined in `CONTRIBUTING.md`.
4. Fix color-related issues in the current PR.
5. Address the line-by-line comments in the PR review.

Once these changes are made, please update the PR. I'll be happy to review it again.

Thank you for your hard work and contribution to CampusUnify!

**Comment by kumartushar0605** [comment]
@PorePranav I have made the changes what you have said so , please merge it

**Review by PorePranav — CHANGES_REQUESTED** [review]
There are still some issues, sorry, but I can't merge this PR right now!


image

Here, kindly fix the layout

image

Dark mode doesn't work on signup, login and profile pages, I get white text over white background:

image

Also, there are some weird background colors present in light mode, as you can see here:

image

If you could fix these issues, I would be more than happy to merge your PR, great work so far! Thank you for your contributions!

File: client/src/pages/Cart.jsx
Comment (Adebesin-Cell) [inline client/src/pages/Cart.jsx:24]: Hey @kumartushar0605,

This is a great start! I noticed that we're using conditional classes to check for dark mode and apply the appropriate styles. I would recommend simplifying this by leveraging Tailwind's built-in dark mode utility classes. For example, instead of conditional logic, you could use:

```suggestion
'dark:bg-gray-800 dark:text-white bg-gray-100 text-black'
```

Tailwind's dark mode utilities make it easier to manage dark mode styles without needing extra JavaScript or conditionals, improving readability and maintainability. Plus, it keeps everything consistent across components.

You can find more information on Tailwind's dark mode utilities here: [Tailwind Dark Mode Docs](https://tailwindcss.com/docs/dark-mode).

Let me know what you think!

cc: @PorePranav

**Review by Adebesin-Cell — CHANGES_REQUESTED** [review]

File: client/src/pages/Cart.jsx
Comment (PorePranav) [reply client/src/pages/Cart.jsx:24]: Thank you for your inputs. This is exactly what I addressed in the first PR review. If you are interested to work on this along with @kumartushar0605 you guys can collaborate. I'll make sure that both of you get the credit!

Comment (Adebesin-Cell) [reply client/src/pages/Cart.jsx:24]: Okay, cool. I’m interested with working to fix the issues.
Make sense, thank you.

Comment (Adebesin-Cell) [reply client/src/pages/Cart.jsx:24]: Do I create a fork of the repo? Or perhaps it’s based off creating branch off main?

Comment (PorePranav) [reply client/src/pages/Cart.jsx:24]: How about I pull @kumartushar0605 's PR to dev branch, and then you pull it from there to fix the changes

> Do I create a fork of the repo? Or perhaps it’s based off creating branch off main?

Comment (Adebesin-Cell) [reply client/src/pages/Cart.jsx:24]: > How about I pull @kumartushar0605 's PR to dev branch, and then you pull it from there to fix the changes
> 
> > Do I create a fork of the repo? Or perhaps it’s based off creating branch off main?

That makes sense, appreciate that. Thanks, what’s the dev branch name?

Comment (PorePranav) [reply client/src/pages/Cart.jsx:24]: > > How about I pull @kumartushar0605 's PR to dev branch, and then you pull it from there to fix the changes
> > > Do I create a fork of the repo? Or perhaps it’s based off creating branch off main?
> 
> That makes sense, appreciate that. Thanks, what’s the dev branch name?

I have merged this PR into dev branch of the main repository, kindly pull from there and you can then start your work. Thank you!

**Comment by PorePranav** [comment]
@kumartushar0605 I have merged your PR, thank you for your contributions!

**Comment by Adebesin-Cell** [comment]
Hi @PorePranav, I hope you're doing well! I was wondering if I should be sourcing the environment variables myself, or if they are provided for the project? Thanks in advance!

**Comment by Adebesin-Cell** [comment]
Follow up question, do you have styling guide for the dark mode?

**Comment by PorePranav** [comment]
> Hi @PorePranav, I hope you're doing well! I was wondering if I should be sourcing the environment variables myself, or if they are provided for the project? Thanks in advance!


Thanks for reaching out! For the project, you’ll need to implement your own environment variables. Since you're focusing on the dark mode implementation, the only environment variables you’ll need to set up for the backend are the database URL and the JWT-related variables.

For your reference, you can find examples of the environment configuration in the repository under `.env.example`. This should give you a clear idea of what’s required.

As for your question about a styling guide for dark mode, there isn't a specific guide. However, I recommend using colors that are related to the ones already defined in the Tailwind config as a reference. Feel free to unleash your creativity and have fun with it!

If you have any further questions or need assistance, don’t hesitate to ask!

Thank you!

**Comment by PorePranav** [comment]
Also, let's continue the conversion on issue #6 rather than here on the PR

**Comment by Adebesin-Cell** [comment]
> > Hi @PorePranav, I hope you're doing well! I was wondering if I should be sourcing the environment variables myself, or if they are provided for the project? Thanks in advance!
> 
> Thanks for reaching out! For the project, you’ll need to implement your own environment variables. Since you're focusing on the dark mode implementation, the only environment variables you’ll need to set up for the backend are the database URL and the JWT-related variables.
> 
> For your reference, you can find examples of the environment configuration in the repository under `.env.example`. This should give you a clear idea of what’s required.
> 
> As for your question about a styling guide for dark mode, there isn't a specific guide. However, I recommend using colors that are related to the ones already defined in the Tailwind config as a reference. Feel free to unleash your creativity and have fun with it!
> 
> If you have any further questions or need assistance, don’t hesitate to ask!
> 
> Thank you!


Hi @PorePranav,

Thank you so much for your detailed response! Got it, I'll go ahead and implement the environment variables accordingly. I appreciate the heads-up regarding the dark mode design as well. I’ll try to experiment with the colors based on the Tailwind config, though design isn't really my strong suit. I’ll see what I can come up with!

Thanks again for your guidance! If I have any other questions, I’ll be sure to reach out.

**Comment by Adebesin-Cell** [comment]
> Also, let's continue the conversion on issue #6 rather than here on the PR

Alright, thank you 🙏

**Pull request merged**
